src/stream_type.rs

```diff
@@ -52,4 +54,38 @@ impl io::Write for StreamType {
     }
 }
 
-impl Io for StreamType {}
+impl AsyncRead for StreamType {
+    unsafe fn prepare_uninitialized_buffer(&self, buf: &mut [u8]) -> bool {
```

Comment:
Yeah there was a *huge* performance difference. Like, I think throughput doubled or something crazy with this implemented. The problem is that otherwise the message buffer is always zeroed, which matters a lot for large buffer sizes like in the throughput benchmark. The basic idea, which I'll add a comment for, is that by overriding the default for this fn, `StreamType` is obliged to never read the uninitialized buffer (which it doesn't).